## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 448371`
Commit: `ea9de94bcf82e11dd5523beb15ac7a64e9794fe9`

---
### `aarch64-darwin`
<details>
  <summary>:x: 21 packages failed to build:</summary>
  <ul>
    <li>beets (beetsPackages.beets, beetsPackages.beets-stable, python313Packages.beets)</li>
    <li>beets.dist (beetsPackages.beets-stable.dist, beetsPackages.beets.dist, python313Packages.beets.dist)</li>
    <li>beets.doc (beetsPackages.beets-stable.doc, beetsPackages.beets.doc, python313Packages.beets.doc)</li>
    <li>beets.man (beetsPackages.beets-stable.man, beetsPackages.beets.man, python313Packages.beets.man)</li>
    <li>beetsPackages.alternatives</li>
    <li>beetsPackages.alternatives.dist</li>
    <li>beetsPackages.audible</li>
    <li>beetsPackages.audible.dist</li>
    <li>beetsPackages.beets-minimal</li>
    <li>beetsPackages.beets-minimal.dist</li>
    <li>beetsPackages.beets-minimal.doc</li>
    <li>beetsPackages.beets-minimal.man</li>
    <li>footage</li>
    <li>python312Packages.beetcamp</li>
    <li>python312Packages.beetcamp.dist</li>
    <li>python312Packages.beets</li>
    <li>python312Packages.beets.dist</li>
    <li>python312Packages.beets.doc</li>
    <li>python312Packages.beets.man</li>
    <li>python313Packages.beetcamp</li>
    <li>python313Packages.beetcamp.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 54 packages built:</summary>
  <ul>
    <li>cavalcade</li>
    <li>cavalcade.dist</li>
    <li>gst_all_1.gst-plugins-rs</li>
    <li>gst_all_1.gst-plugins-rs.dev</li>
    <li>mediagoblin</li>
    <li>mediagoblin.dist</li>
    <li>mopidy-bandcamp</li>
    <li>mopidy-bandcamp.dist</li>
    <li>mopidy-iris</li>
    <li>mopidy-iris.dist</li>
    <li>mopidy-jellyfin</li>
    <li>mopidy-jellyfin.dist</li>
    <li>mopidy-listenbrainz</li>
    <li>mopidy-listenbrainz.dist</li>
    <li>mopidy-local</li>
    <li>mopidy-local.dist</li>
    <li>mopidy-mopify</li>
    <li>mopidy-mopify.dist</li>
    <li>mopidy-mpd</li>
    <li>mopidy-mpd.dist</li>
    <li>mopidy-mpris</li>
    <li>mopidy-mpris.dist</li>
    <li>mopidy-muse</li>
    <li>mopidy-muse.dist</li>
    <li>mopidy-musicbox-webclient</li>
    <li>mopidy-musicbox-webclient.dist</li>
    <li>mopidy-notify</li>
    <li>mopidy-notify.dist</li>
    <li>mopidy-podcast</li>
    <li>mopidy-podcast.dist</li>
    <li>mopidy-scrobbler</li>
    <li>mopidy-scrobbler.dist</li>
    <li>mopidy-somafm</li>
    <li>mopidy-somafm.dist</li>
    <li>mopidy-soundcloud</li>
    <li>mopidy-soundcloud.dist</li>
    <li>mopidy-spotify</li>
    <li>mopidy-spotify.dist</li>
    <li>mopidy-subidy</li>
    <li>mopidy-subidy.dist</li>
    <li>mopidy-tidal</li>
    <li>mopidy-tidal.dist</li>
    <li>mopidy-tunein</li>
    <li>mopidy-tunein.dist</li>
    <li>mopidy-youtube</li>
    <li>mopidy-youtube.dist</li>
    <li>mopidy-ytmusic</li>
    <li>mopidy-ytmusic.dist</li>
    <li>muzika</li>
    <li>python312Packages.gst-python</li>
    <li>python312Packages.gst-python.dev</li>
    <li>python313Packages.gst-python</li>
    <li>python313Packages.gst-python.dev</li>
    <li>showtime</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>beets</summary>
<pre>checking for references to /nix/var/nix/builds/nix-build-beets-2.4.0.drv-2/b/ in /nix/store/1zflc1j7gakmb8r7c58cyhavzhhh38cs-beets-2.4.0-dist...
patching script interpreter paths in /nix/store/1zflc1j7gakmb8r7c58cyhavzhhh38cs-beets-2.4.0-dist
Rewriting #!/nix/store/i8r431v9g2hwh9g0kkp2lrgd426szgnq-python3-3.13.7/bin/python3.13 to #!/nix/store/i8r431v9g2hwh9g0kkp2lrgd426szgnq-python3-3.13.7
wrapping `/nix/store/2x6yf7vgcr3liph10kd93gixih1swsmc-beets-2.4.0/bin/beet&#x27;...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;installCheckPhase&quot; }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonCatchConflictsPhase&quot; }
Running phase: pythonRemoveBinBytecodePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonRemoveBinBytecodePhase&quot; }
Running phase: pythonImportsCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonImportsCheckPhase&quot; }
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pytestCheckPhase&quot; }
Executing pytestCheckPhase
Disabled tests path glob &quot;test/plugins/test_player.py&quot; does not match any paths. Aborting</pre>
</details>
<details>
<summary>beetsPackages.beets-minimal</summary>
<pre>checking for references to /nix/var/nix/builds/nix-build-beets-2.4.0.drv-0/b/ in /nix/store/wwaqc92riildn8b3k7kg0h32y2cgfc8y-beets-2.4.0-dist...
patching script interpreter paths in /nix/store/wwaqc92riildn8b3k7kg0h32y2cgfc8y-beets-2.4.0-dist
Rewriting #!/nix/store/i8r431v9g2hwh9g0kkp2lrgd426szgnq-python3-3.13.7/bin/python3.13 to #!/nix/store/i8r431v9g2hwh9g0kkp2lrgd426szgnq-python3-3.13.7
wrapping `/nix/store/2njlx5jsk9nqa9nigj30dc43wlzbhm0s-beets-2.4.0/bin/beet&#x27;...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;installCheckPhase&quot; }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonCatchConflictsPhase&quot; }
Running phase: pythonRemoveBinBytecodePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonRemoveBinBytecodePhase&quot; }
Running phase: pythonImportsCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonImportsCheckPhase&quot; }
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pytestCheckPhase&quot; }
Executing pytestCheckPhase
Disabled tests path glob &quot;test/plugins/test_player.py&quot; does not match any paths. Aborting</pre>
</details>
<details>
<summary>footage</summary>
<pre>  make[5]: *** Waiting for unfinished jobs....
  /nix/var/nix/builds/nix-build-footage-1.3.2.drv-0/b/tbef2-0/gettext/libtextstyle/lib/tputs.c:29:28: warning: unused parameter &#x27;affcnt&#x27; [-Wunused-parameter]
     29 | tputs (const char *cp, int affcnt, int (*outcharfun) (int))
        |                            ^
  1 warning generated.
  make[4]: *** [Makefile:2157: all] Error 2
  make[3]: *** [Makefile:1786: all-recursive] Error 1
  make[2]: *** [Makefile:1714: all] Error 2
  make[1]: *** [Makefile:403: all-recursive] Error 1
  make: *** [Makefile:359: all] Error 2

  thread &#x27;main&#x27; panicked at /nix/var/nix/builds/nix-build-footage-1.3.2.drv-0/b/footage-1.3.2-vendor/gettext-sys-0.21.3/build.rs:300:5:

  command did not execute successfully, got: exit status: 2

  build script failed, must exit now
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
FAILED: [code=101] src/footage 
/nix/store/wbs75n9ffs3ac1pmn4fc7wqs0m9r1z3c-coreutils-9.7/bin/env CARGO_HOME=/nix/var/nix/builds/nix-build-footage-1.3.2.drv-0/b/source/build/cargo-home /nix/store/gzv6psv17kv7x9s01w8jhi0h2cg6z15p-cargo-1.89.0/bin/cargo build --manifest-path /nix/var/nix/builds/nix-build-footage-1.3.2.drv-0/b/source/Cargo.toml --target-dir /nix/var/nix/builds/nix-build-footage-1.3.2.drv-0/b/source/build/src --release &amp;&amp; cp src/release/footage src/footage
ninja: build stopped: subcommand failed.</pre>
</details>
<details>
<summary>python312Packages.beets</summary>
<pre>checking for references to /nix/var/nix/builds/nix-build-beets-2.4.0.drv-1/b/ in /nix/store/3v9rsahmygzzwg33gvg68xgmawxg160f-beets-2.4.0-dist...
patching script interpreter paths in /nix/store/3v9rsahmygzzwg33gvg68xgmawxg160f-beets-2.4.0-dist
Rewriting #!/nix/store/y6hx9cmhnnypdwqy8i5knrzk9lx88mwv-python3-3.12.11/bin/python3.12 to #!/nix/store/y6hx9cmhnnypdwqy8i5knrzk9lx88mwv-python3-3.12.11
wrapping `/nix/store/z29hadvv936ywbk9bra5virgb8k5bxhy-beets-2.4.0/bin/beet&#x27;...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;installCheckPhase&quot; }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonCatchConflictsPhase&quot; }
Running phase: pythonRemoveBinBytecodePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonRemoveBinBytecodePhase&quot; }
Running phase: pythonImportsCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonImportsCheckPhase&quot; }
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pytestCheckPhase&quot; }
Executing pytestCheckPhase
Disabled tests path glob &quot;test/plugins/test_player.py&quot; does not match any paths. Aborting</pre>
</details>
</details>

--- a/pkgs/development/libraries/gstreamer/rs/default.nix
+++ b/pkgs/development/libraries/gstreamer/rs/default.nix
@@ -116,6 +116,7 @@ let
         ++ lib.optionals stdenv.hostPlatform.isDarwin [
           "reqwest" # tests hang on darwin
           "threadshare" # tests cannot bind to localhost on darwin
+          "uriplaylistbin" # thread reqwest-internal-sync-runtime attempred to create a NULL object (in test_cache)
           "webp" # not supported on darwin (upstream crate issue)
         ]
         ++ lib.optionals (!gst-plugins-base.glEnabled || !withGtkPlugins) [

--- a/pkgs/development/python-modules/gst-python/default.nix
+++ b/pkgs/development/python-modules/gst-python/default.nix
@@ -87,10 +87,10 @@ buildPythonPackage rec {
     updateScript = directoryListingUpdater { };
   };
 
-  meta = with lib; {
+  meta = {
     homepage = "https://gstreamer.freedesktop.org";
     description = "Python bindings for GStreamer";
-    license = licenses.lgpl2Plus;
+    license = lib.licenses.lgpl2Plus;
     maintainers = [ ];
   };
 }

--- a/pkgs/development/python-modules/gst-python/default.nix
+++ b/pkgs/development/python-modules/gst-python/default.nix
@@ -42,6 +42,13 @@ buildPythonPackage rec {
     })
   ];
 
+  postPatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    # The analytics tests often timeout under load on Darwin (e.g. on Hydra), so remove them
+    substituteInPlace testsuite/meson.build --replace-fail \
+      "['Test analytics', 'test_analytics.py', ['gst-plugins-bad/gst-libs/gst/analytics', 'gst-plugins-base/gst-libs/gst/video']]," \
+      ""
+  '';
+
   # Python 2.x is not supported.
   disabled = !isPy3k;
 

